src: split out shared functionality to common-priv.h.

Ionic opened the pull request:
Less code duplication.

This header file lives in `src/` because it's not meant to be installed. It's fully private.

While that may not sound like a big deal yet, we'll use this more extensively soon. Trust me, it's a good thing.

As an added benefit, we can also consolidate `__pthread_fchdir()` and `best_fchdir`, which were exactly the same. Yay.

kencu commented:
pthread_fchdir doesn’t exist on tiger so that’s why we had best_fchdir.

If someone removed that when I wasn’t looking, that’s probably why we had all those recent errors on Tiger I couldn’t explain. 

I’ll have to put it back.

Ionic commented:
Rebased against current master.

No, nobody removed it. But we've had two functions doing exactly the same -- `best_fchdir` (in its own compilation unit `src/best_fchdir.c`) and `__pthread_fchdir` (local to `src/atcalls.c`). I consolidated both to use `best_fchdir` instead and removed the duplicated one in `src/atcalls.c`.

Ionic commented:
Hm... actually, the defines are getting complicated, let's redo this a bit.

Ionic commented:
That looks way better/less complicated and fixes a bug in my implementation that I haven't realized before due to added consistency when it comes to when the `ATCALL` macros or `best_fchdir` are actually needed.

Ionic commented:
And then, because I cannot do without typos, a typo fix.

michaelld commented on src/common-priv.h:
```diff
@@ -0,0 +1,77 @@
+/*-
+ * Copyright (c) 2019 Ken Cunningham kencu@macports.org
```
Need to update this (C) ... either yourself or MP & the new year

Ionic replied:
I intentionally left it as-is, mostly because this is just a refactor/move of code without actual changes.

But yeah, the minor changes in preprocessor conditions could justify an update, so I'll do that.

michaelld replied:
You are creating a new file, which needs a new (C): either totally new or keep the old (C) for the code as copied, but update it for anything new.

Ionic replied:
Updated, with interval to this year.

michaelld replied:
works for me; thx!

michaelld commented on src/common-priv.h:
```diff
@@ -0,0 +1,77 @@
+#ifndef SYS___pthread_fchdir
+# define SYS___pthread_fchdir 349
+#endif
+
```
extra \n to be removed

Ionic replied:
That, likewise, is intentional. I like to group logical things via double newlines and find that the visual cues make reading code a bit easier. Personal preference, though.

If I remove it, I'd better also remove the other double newline for consistency.

michaelld replied:
Hmmm ... I don't think we're consistent in this spacing. How about we just leave it as you've done it, since that's less work :)

michaelld reviewed (changes requested):
A couple minor nit-picks ... otherwise LGTM & makes sense to provide this functionality in a common but private header

michaelld reviewed (approved):
LGTM

michaelld commented:
I'll leave this up to @kencu now ... I think it's a good change!
